importer/api/__init__.py

```diff
@@ -36,17 +37,22 @@ def import_course_from_file(filename, user_id):
     except ArchiveException:
         remove(filename)
         raise ValueError("Invalid OLX archive, unable to extract.")
-    dirs = glob(join(tempdir, "*"))
-    if len(dirs) != 1:
-        rmtree(tempdir)
-        remove(filename)
-        raise ValueError("Invalid OLX archive, bad directory structure.")
-    import_course_from_path(dirs[0], user_id)
+    course_imported = False
```

Comment:
So the comment could be: "A valid OLX archive has a single occurrence of the file `course.xml` in its root directory, or no `course.xml` in its root and a single occurrence of `course.xml` in one or more of the root directory's children." 

I've shied away from adding many comments since it's so easy for comments to become out of sync with the code they describe.